jscomp/others/js_exn.ml

```diff
@@ -45,6 +45,12 @@ external makeError : string -> error = "Error" [@@bs.new]
 external isCamlExceptionOrOpenVariant : 
   'a -> bool = "caml_is_extension"
 
+let unsafeAnyToExn obj =
+  if isCamlExceptionOrOpenVariant obj then
+    (Obj.magic obj : exn)
+  else
+    Error ((Obj.magic obj) : t)
```

Comment:
think twice, this is a safe function. but it is encouraged only for internal usage, maybe we can rename it into `anyToExnInternal`

Reply:
what is considered to be internal? do you mean i should create some function in `caml_js_exceptions`, and use an external to bind to it? I browsed the code yesterday, but didn't really understand how one defines a "real internal" function.

> maybe we can rename it into anyToExnInternal

Yes, it's not really unsafe, so I'd like this name change